src/PixiEditor/ViewModels/SubViewModels/Main/LayersViewModel.cs

```diff
@@ -48,10 +53,80 @@ public bool CanDeleteSelected()
     {
-        var member = Owner.DocumentManagerSubViewModel.ActiveDocument?.SelectedStructureMember;
+        var doc = Owner.DocumentManagerSubViewModel?.ActiveDocument;
+
+        var member = doc.SelectedStructureMember;
         if (member is null)
             return;
+
         member.Document.Operations.DeleteStructureMember(member.GuidValue);
+        var allLayers = doc.StructureHelper.GetAllMembers().ToArray();
+
+        //Declare the layer that will be selected
+        StructureMemberViewModel baseLayer = member;
+
+        //Checking if the length is greater than one, to avoid crashing.
+        if (allLayers.Length > 0)
+        {  
+            for (int i = 0; i < allLayers.Length; i++)
+            {
+                //Finding the layer that is being deleted
+                if (allLayers[i] == member)
+                {
+                    //Selecting the new layer
+                    if (i > 0)
+                    {
+                        baseLayer = allLayers[i - 1];
+                    }
+                    else if (i == allLayers.Length - 1)
+                    {
+                        baseLayer = allLayers[i];
+                    }
+                    else if (i == 0)
+                    {
+                        baseLayer = allLayers[i + 1];
+                    }
+
+                    break;
+                }
+                else
+                {
+                    //Managing elements inside folders
+                    if (allLayers[i] is FolderViewModel folder)
+                    {
+                        var list = folder.Children.OfType<StructureMemberViewModel>().ToArray();
+
+                        if (list.Length > 1)
+                        {
+                            for (int j = 0; j < list.Length; j++)
+                            {
+                                //Finding the layer that is being deleted
+                                if (list[j] == member)
+                                {
+                                    // Selecting the new layer
+                                    if (j > 0)
+                                    {
+                                        baseLayer = list[j - 1];
+                                    }
+                                    else if (j == list.Length - 1)
+                                    {
+                                        baseLayer = list[j];
+                                    }
+                                    else if (j == 0)
+                                    {
+                                        baseLayer = list[j + 1];
+                                    }
+                                    break; 
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            doc.InternalSetSelectedMember(baseLayer);
+            doc.Operations.AddSoftSelectedMember(baseLayer.GuidValue);
```

Comment:
"Soft" selection is used in the case where you have multiple layers selected. To handle drawing in that case, we need to choose a single layer to draw on. Because of that there are two types of selection, "Hard" selection (only one layer can be hard-selected) and "Soft" selection (any selected layers that aren't the main one will be "soft" selected).

It looks like you are trying to hard-select `baseLayer`, and immediately afterwards also soft-select it, which doesn't make sense (and there is a check in `DocumentUpdater.ProcessAddSoftSelectedMember` which ensures that you can't soft-select a hard-selected layer)